### PR TITLE
Grid multiple media

### PR DESCRIPTION
**Pull request by kgcreative**
Addresses #605 - Extend grid-media() to accept multiple maps.

- [x] Commit message has a short title & issue references
- [x] Commits are squashed
- [x] The build will pass (run `bundle exec rake`)

### CODE REVIEW

File: core/neat/mixins/_grid-media.scss [hunk 1]
```diff
@@ -46,22 +52,32 @@
...
+///       width: calc(50% - 75px);
+///       float: left;
+///       margin-left: 70px;
+///     }
+///   }
 
-@mixin grid-media($grid) {
-  $_media: _retrieve-neat-setting($grid, media);
-  $_query: _neat-parse-media($_media);
+@mixin grid-media($grids...) {
```
Comment (whmii) [inline core/neat/mixins/_grid-media.scss]: Ive never seen this syntax, is there documentation for this?

**Comment by whmii** [comment]
Make sure you try it w/ only one map passed in to the mixin. If i recall sass is really crappy about distinguishing between maps and lists of maps. If you run a for each on a map it may break apart the map and try to run on each of the maps properties.

File: core/neat/mixins/_grid-media.scss (hunk 1 above)
Comment (kgcreative) [inline core/neat/mixins/_grid-media.scss]: http://sass-lang.com/documentation/file.SASS_REFERENCE.html#variable_arguments

the '...' syntax is used to specify an unknown number of variable arguments.

**Comment by kgcreative** [comment]
@whmii - I've successfully used this on my own repo, for both use cases where i'm passing an inline map, a single map, or multiple maps.

**Comment by whmii** [comment]
OIC, I'll run this Mañana and check it out

**Comment by kgcreative** [comment]
bump!

**Comment by whmii** [comment]
@kgcreative Alrighty! Rebase & squash to a single commit and we're good. LGTM

**Comment by kgcreative** [comment]
@whmii - uhh, I think that's about as squashed as I can get it, but git often confuses me. 🤔

**Comment by whmii** [comment]
We can ignore the hound comments here. most of these are more for sites than for libraries

File: core/neat/mixins/_grid-media.scss [hunk 2]
```diff
@@ -13,28 +13,39 @@
...
 ///
+///   If multiple grids are passed in to `grid-media`, it will loop through the
+///   contents of the block, quickly allowing you to apply multiple grids in a
+///   single mixin.
+///
 /// @example scss
 ///   $custom-neat-grid: (
 ///     columns: 12,
 ///     gutter: 50px,
+///     media: 'screen and (max-width: 999px)',
```
Comment (tysongach) [inline core/neat/mixins/_grid-media.scss]: Swap single quotes for double.

File: core/neat/mixins/_grid-media.scss [hunk 3]
```diff
@@ -46,22 +57,32 @@
...
+///       width: calc(50% - 75px);
+///       float: left;
+///       margin-left: 70px;
+///     }
+///   }
 
-@mixin grid-media($grid) {
-  $_media: _retrieve-neat-setting($grid, media);
-  $_query: _neat-parse-media($_media);
+@mixin grid-media($grid...) {
```
Comment (tysongach) [inline core/neat/mixins/_grid-media.scss]: Even though this argument name is the same, it seems that changing it to an arglist still creates a breaking change. I threw together [a simple CodePen](https://codepen.io/tysongach/pen/prNpQM?editors=0100) and if I call the mixin (as I would in its current 2.1 version) using a named argument, swapping the mixin to the arglist throws an error:

> Argument $grid of mixin grid-media cannot be used as a named argument.

If that’s the case, I don’t believe we’d be able to ship in a minor release; would need a major release up to 3.0.

**Pull request merged**